src/cargo/lib.rs

```diff
@@ -1,7 +1,7 @@
 // For various reasons, some idioms are still allow'ed, but we would like to
 // test and enforce them.
 #![warn(rust_2018_idioms)]
-#![cfg_attr(test, deny(warnings))]
+// #![cfg_attr(test, deny(warnings))]
```

Comment:
This needs resolving

Reply:
Ugh, a great catch. This should not have been committed and I had no safeguards to prevent me from doing that 🤦.
It's fixed by rewriting the PR that accidentally introduced it, please let me know if I rather should avoid rewriting, or do it later during cleanup.

Thinking about it, even though this git history was meant to be useful and kept, maybe in the end it's best to squash it as it's already questionable whether I am succeeding with this goal.

Reply:
This is fixed in d92d04840c1fb518faf8f964054b48f64e4920db.

Reply:
> Ugh, a great catch. This should not have been committed and I had no safeguards to prevent me from doing that 🤦.

imo this is system problem and not a user problem, so I [posted a way of solving this for everyone](https://rust-lang.zulipchat.com/#narrow/stream/246057-t-cargo/topic/.60.23!.5Bcfg_attr.28test.2C.20deny.28warnings.29.29.5D.60).